sway-core/src/ir_generation/function.rs

```diff
@@ -57,6 +57,36 @@ use std::collections::HashMap;
 /// forced, into a value if that is desired. All the temporary values are manipulated with simple
 /// loads and stores, rather than anything more complicated like `mem_copy`s.
 
+// Wrapper around Value to quickly distinguish between diverging and non-diverging values.
+enum ValueDivergence {
+    Diverging(Value),
+    NotDiverging(Value),
+}
+
+impl ValueDivergence {
+    pub fn mk_value_divergence(value: Value, context: &Context) -> Self {
+        if value.is_diverging(context) {
+            Self::Diverging(value)
+        } else {
+            Self::NotDiverging(value)
+        }
+    }
+
+    pub fn is_diverging(&self) -> bool {
+        matches!(self, Self::Diverging(_))
+    }
+
+    pub fn get_value(&self) -> Value {
```

Comment:
In Rust, getters usually don't use the word `get` because there's no ambiguity to remove.

This should be called `value` and `get_type` should be called `type`.